### PR TITLE
Make random Aegislash sets have King's Shield

### PR DESCRIPTION
**Pull request by Relados**
Any viable set for Aegislash should have King's Shield, since not having this would render it's ability useless and make it way too frail after one attacking move.

### CODE REVIEW

File: data/scripts.js [hunk 1]
```diff
@@ -1402,6 +1402,25 @@ exports.BattleScripts = {
 			moves[3] = toId(template.requiredMove);
 			hasMove[toId(template.requiredMove)] = true;
 		}
+		
+		// all viable aegislash sets should have king's shield
+		if (template.species === 'Aegislash' && !hasMove['kingsshield']) {
+			// replace a non-attacking move if the set has it
+			for (var i = 0; i < moves.length; ++i) {
+				if (moves[i].category === 'Status') {
```
Comment (ascriptmaster) [inline data/scripts.js]: This doesn't work. You have to call `Tools#getMove` on the move in order to check its category. `moves[i]` is just a string.

**Comment by kotarou3** [comment]
Wouldn't it be better to fill Aegislash's first slot with King's Shield **before** the rest of the moves are generated?

**Comment by ascriptmaster** [comment]
Putting King's Shield in the first slot still gives it a chance of being rejected.

For example, Aegislash can roll up the set `['kingsshield', 'swordsdance', 'autonomize', 'shadowsneak']`. Since it has Swords Dance, the algorithm will mark it as `setupType = 'Physical'`.

Then, the algorithm will reach line 1314, which reads:

	if (setupType === 'Physical' && move.category !== 'Physical' && counter['Physical'] < 2) {
		rejected = true;
	}

Since King's Shield is not a Physical move, and there is only one Physical move on the set, it will be rejected!

There's a reason we replace moves with our "required moves" after we are done processing the moveset as needed. It's because otherwise our required move still has a small chance of being rejected!

File: data/scripts.js (hunk 1 above)
Comment (Joimer) [inline data/scripts.js]: Also why ++i instead of i++ :p

**Comment by Joimer** [comment]
btw, I think what's making the build fail is a newline with tabs and nothing else.
Also I'd check what moves are you getting out before getting them out, say it's physical with swords dance and you get rid of SD... not cool bro.

**Comment by TheImmortal** [comment]
It's only Status moves are Swords Dance and Autotomize. It should never replace Swords Dance.

**Comment by TheImmortal** [comment]
66615c5331daef6068bff4b13181f7f235da9678 should help it get King's Shield most, if not all, of the time.

**Pull request closed without merging**